src/streaming/controllers/BufferController.js

```diff
@@ -400,38 +303,32 @@ function BufferController(config) {
 
     function notifyBufferStateChanged(state) {
         if (bufferState === state || (type === 'fragmentedText' && textSourceBuffer.getAllTracksAreDisabled())) return;
-
         bufferState = state;
         addBufferMetrics();
         eventBus.trigger(Events.BUFFER_LEVEL_STATE_CHANGED, {sender: instance, state: state, mediaType: type, streamInfo: streamProcessor.getStreamInfo()});
-        let eventType = state === BUFFER_LOADED ? Events.BUFFER_LOADED : Events.BUFFER_EMPTY;
-        eventBus.trigger(eventType, {mediaType: type});
-        log(state === BUFFER_LOADED ? ('Got enough buffer to start.') : ('Waiting for more buffer before starting playback.'));
+        eventBus.trigger(state === BUFFER_LOADED ? Events.BUFFER_LOADED : Events.BUFFER_EMPTY, {mediaType: type});
+        log(state === BUFFER_LOADED ? 'Got enough buffer to start.' : 'Waiting for more buffer before starting playback.');
     }
 
 
     function handleInbandEvents(data, request, mediaInbandEvents, trackInbandEvents) {
-        var fragmentStarttime = Math.max(isNaN(request.startTime) ? 0 : request.startTime, 0);
-        var eventStreams = [];
-        var events = [];
 
-        var eventBoxes,
-            event,
-            isoFile,
-            inbandEvents;
+        const fragmentStartTime = Math.max(isNaN(request.startTime) ? 0 : request.startTime, 0);
+        const eventStreams = [];
+        const events = [];
 
-        inbandEventFound = false;
+        inbandEventFound = false; //TODO Discuss why this is hear!
```

Comment:
See https://github.com/Dash-Industry-Forum/dash.js/issues/853

I have a major refactor of event message handling which I hope to land next month when Firefox 49 ships. Much of this stuff is removed. See https://github.com/Dash-Industry-Forum/dash.js/issues/1164 for more info.